pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -261,60 +203,47 @@ func (t ResolvedPipelineTask) isCancelled() bool {
 			atLeastOneCancelled = atLeastOneCancelled || runCancelled
 		}
 		return atLeastOneCancelled && isDone
-	case t.IsCustomTask():
-		if t.RunObject == nil {
-			return false
-		}
-		c := t.RunObject.GetStatusCondition().GetCondition(apis.ConditionSucceeded)
-		return c != nil && c.IsFalse() && c.Reason == v1beta1.CustomRunReasonCancelled.String()
-	case t.PipelineTask.IsMatrixed():
-		if len(t.TaskRuns) == 0 {
-			return false
-		}
-		isDone := true
-		atLeastOneCancelled := false
-		for _, taskRun := range t.TaskRuns {
-			isDone = isDone && taskRun.IsDone()
-			c := taskRun.Status.GetCondition(apis.ConditionSucceeded)
-			taskRunCancelled := c.IsFalse() && c.Reason == v1beta1.TaskRunReasonCancelled.String()
-			atLeastOneCancelled = atLeastOneCancelled || taskRunCancelled
-		}
-		return atLeastOneCancelled && isDone
-	default:
-		if t.TaskRun == nil {
-			return false
-		}
-		c := t.TaskRun.Status.GetCondition(apis.ConditionSucceeded)
-		return c != nil && c.IsFalse() && c.Reason == v1beta1.TaskRunReasonCancelled.String()
 	}
+	if len(t.TaskRuns) == 0 {
+		return false
+	}
+	isDone := true
+	atLeastOneCancelled := false
+	for _, taskRun := range t.TaskRuns {
+		isDone = isDone && taskRun.IsDone()
+		c := taskRun.Status.GetCondition(apis.ConditionSucceeded)
+		taskRunCancelled := c.IsFalse() && c.Reason == v1beta1.TaskRunReasonCancelled.String()
+		atLeastOneCancelled = atLeastOneCancelled || taskRunCancelled
+	}
+	return atLeastOneCancelled && isDone
 }
 
 // isScheduled returns true when the PipelineRunTask itself has a TaskRun
 // or Run associated.
 func (t ResolvedPipelineTask) isScheduled() bool {
 	if t.IsCustomTask() {
-		return t.RunObject != nil
+		return len(t.RunObjects) > 0
 	}
-	return t.TaskRun != nil
+	return len(t.TaskRuns) > 0
 }
 
 // isStarted returns true only if the PipelineRunTask itself has a TaskRun or
 // Run associated that has a Succeeded-type condition.
 func (t ResolvedPipelineTask) isStarted() bool {
 	if t.IsCustomTask() {
-		return t.RunObject != nil && t.RunObject.GetStatusCondition().GetCondition(apis.ConditionSucceeded) != nil
+		return len(t.RunObjects) > 0 && t.RunObjects[0].GetStatusCondition().GetCondition(apis.ConditionSucceeded) != nil
 	}
-	return t.TaskRun != nil && t.TaskRun.Status.GetCondition(apis.ConditionSucceeded) != nil
+	return len(t.TaskRuns) > 0 && t.TaskRuns[0].Status.GetCondition(apis.ConditionSucceeded) != nil
 }
 
 // isConditionStatusFalse returns true when a task has succeeded condition with status set to false
 // it includes task failed after retries are exhausted, cancelled tasks, and time outs
 func (t ResolvedPipelineTask) isConditionStatusFalse() bool {
 	if t.isStarted() {
 		if t.IsCustomTask() {
-			return t.RunObject.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsFalse()
+			return t.RunObjects[0].GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsFalse()
```

Comment:
This doesn't look right: it seems like it will only work for a singular pipeline task. It probably shouldn't be a member function of `ResolvedPipelineTask`, since a `ResolvedPipelineTask` can be associated with multiple TaskRuns/Runs. We should probably just define a helper for TaskRuns/Runs.